Quaternion/Quaternion.py

```diff
@@ -113,34 +126,44 @@ def __init__(self, attitude):
         else:
             # make it an array and check to see if it is a supported shape
             attitude = np.array(attitude)
-            if len(attitude) == 4:
+            if ((attitude.shape == (3, 3) and (intype is None or intype == 'transform'))
+                or (attitude.ndim == 3 and attitude.shape[-1] == 3 and attitude.shape[2] == 3)):
+                    self._set_transform(attitude)
+            elif (intype == 'quaternion' or attitude.shape == (4,) or
+                  (attitude.ndim == 2 and attitude.shape[-1] == 4)):
                 self._set_q(attitude)
-            elif attitude.shape == (3, 3):
-                self._set_transform(attitude)
-            elif attitude.shape == (3,):
+            elif (intype == 'equatorial' or attitude.shape == (3,)
+                  or (attitude.ndim == 2 and attitude.shape[-1] == 3)):
                 self._set_equatorial(attitude)
             else:
                 raise TypeError(
-                    "attitude is not one of possible types (3 or 4 elements, Quat, or 3x3 matrix)")
+                    "attitude is not one of possible types "
+                    "(3 or 4 elements, Quat, or 3x3 matrix, or N x (those types))")
 
     def _set_q(self, q):
         """
         Set the value of the 4 element quaternion vector
+        May be 4 element list or array or N x 4 element array.
 
         :param q: list or array of normalized quaternion elements
         """
         q = np.array(q)
-        if abs(np.sum(q**2) - 1.0) > 1e-6:
+        if q.ndim == 1:
+            q = q[np.newaxis]
```

Comment:
Throughout, this "vectorized" approach tries to take the single-attitude case, convert it to a 1xN with newaxis for most of the internal operations, and then return just the single-attitude again again.  This was both for convenience and backwards compatibility, but we could go other ways.